pkg/apiserver/etcd_config.go

```diff
@@ -110,6 +111,19 @@ func (c completedEtcdConfig) NewServer() (*ServiceCatalogAPIServer, error) {
 		glog.V(4).Infof("Installing API group %v", provider.GroupName())
 		if err := s.GenericAPIServer.InstallAPIGroup(groupInfo); err != nil {
 			glog.Fatalf("Error installing API group %v: %v", provider.GroupName(), err)
+		} else {
+			// we've sucessfully installed, so hook the stopCh to the destroy func of all the sucessfully installed apigroups
+			for _, mappings := range groupInfo.VersionedResourcesStorageMap { // gv to resource mappings
+				for _, storage := range mappings { // resource name (brokers, brokers/status) to backing storage
+					go func() {
+						s, ok := storage.(*registry.Store)
+						if ok {
+							<-stopCh
```

Comment:
If you go back into this stuff for some clean-up, can you add a comment along the lines of:
```
the other end of this channel will close it when we're shutting down,
which results in all of these go funcs becoming unblocked.
```